demo/app/ios/Runner/flutterPlugin.swift

```diff
@@ -403,20 +411,103 @@ public class SwiftWalletSDKPlugin: NSObject, FlutterPlugin {
+                                             details: "WalletSDK interaction is not initialized, call initSDK()"))
+        }
+        
         do {
-            let walletInitiatedOpenID4CI = try walletSDK?.createOpenID4CIWalletIntiatedInteraction(issuerURI: issuerURI)
-            //TODO- VCS part is not implemented yet, this logic needs to be updated to get the create authorization URL
-            try walletInitiatedOpenID4CI?.getSupportedCredentials()
 
+            let walletInitiatedOpenID4CI = try walletSDK.createOpenID4CIWalletIntiatedInteraction(issuerURI: issuerURI)
+            let supportedCredentials = try walletInitiatedOpenID4CI.getSupportedCredentials()
+            
+            var supportedCredentialsList: [Any] = []
+            for index in 0..<supportedCredentials.length() {
+                var typeStrArray = [String]()
+                for i in 0..<(supportedCredentials.atIndex(index)!.types()?.length())!{
+                    let type1 = supportedCredentials.atIndex(index)!.types()?.atIndex(i)
+                    typeStrArray.append(type1!)
+                }
+                
+                let supportedCredentialResp:[String:Any] = [
+                    "format":  supportedCredentials.atIndex(index)!.format(),
+                    "types":   typeStrArray
+                ]
+                supportedCredentialsList.append(supportedCredentialResp)
+            }
+            self.walletInitiatedOpenID4CI = walletInitiatedOpenID4CI
+            result(supportedCredentialsList)
+        
         } catch {
             result(FlutterError.init(code: "NATIVE_ERR",
                                      message: "error while initializing wallet initiated issuance flow",
                                      details: error))
         }
         
     }
-    
+     
+    public func createAuthorizationUrlWalletInitiatedFlow(arguments: Dictionary<String, Any>, result: @escaping FlutterResult){
+        guard let walletInitiatedOpenID4CI = self.walletInitiatedOpenID4CI else{
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while process wallet initiated flow",
+                                             details: "walletInitiatedOpenID4CI interaction is not initialted"))
+        }
+        
+        guard let scopesFromArgs = arguments["scopes"] as? Array<String> else {
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while creating authorization Url in wallet initiated flow",
+                                             details: "Pass scopes as the arguments"))
+        }
+        
+        guard let credentialTypesFromArgs = arguments["credentialTypes"] as? Array<String> else {
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while creating authorization Url in wallet initiated flow",
```

Comment:
This message doesn't seem to be correct - looks like a copy/paste error maybe?

Reply:
Was your intent to have the same error message for all of the errors from this function? If so - you should add/wrap that message at the place where you're calling this method instead of repeating it here